datadog_checks_base/datadog_checks/base/utils/cache_key/config_set.py

```diff
@@ -0,0 +1,54 @@
+# (C) Datadog, Inc. 2025-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from __future__ import annotations
+
+from collections.abc import Collection
+from typing import TYPE_CHECKING
+
+from .base import CacheKey
+
+if TYPE_CHECKING:
+    from typing import Any, Iterable
+
+    from datadog_checks.base import AgentCheck
+
+
+class ConfigSetCacheKey(CacheKey):
+    """
+    Cache key that is derived from a subset of the check's config options.
+
+    When the subset of config options changes, the cache is invalidated.
+    """
+
+    def __init__(
+        self,
+        check: AgentCheck,
+        config_options: Collection[str],
+    ):
+        super().__init__(check)
+        self.config_options = set(config_options)
+        # Config cannot change on the fly, so we can cache the key
+        self.__key: str | None = None
+
+    def base_key(self) -> str:
+        if self.__key is not None:
+            return self.__key
+
+        merged_config = self.check.init_config | self.check.instance
```

Comment:
1. The agent determines the check ID from only the instance AFAIK (double check) so it would be odd to use the full resolved config. More importantly, there are options that integrations use with the same name that do not have the exact same behavior and therefore this would not be correct.
2. As I mentioned in another comment, the `name` option from the instance config should always be included.